users/src/main/java/io/crate/operation/user/UserManagerService.java

```diff
@@ -73,6 +77,56 @@
     }
 
     @Override
+    public Boolean checkPermission(AnalyzedStatement analyzedStatement,
+                                   SessionContext sessionContext) {
+        return permissionVisitor.process(analyzedStatement, sessionContext);
+    }
+
+    private class PermissionVisitor extends AnalyzedStatementVisitor<SessionContext, Boolean> {
+
+        @Override
+        protected Boolean visitAnalyzedStatement(AnalyzedStatement analyzedStatement,
+                                                 SessionContext sessionContext) {
+            return true;
+        }
+
+        @Override
+        protected Boolean visitSelectStatement(SelectAnalyzedStatement analysis,
+                                               SessionContext sessionContext) {
+            QueriedTable queriedTable = (QueriedTable) analysis.relation();
+
+            if (queriedTable.tableRelation().tableInfo().ident().name().equals("users") &&
+                queriedTable.tableRelation().tableInfo().ident().schema().equals("sys") &&
+                !CRATE_USER.name().equals(sessionContext.userName())) {
```

Comment:
i wouldn't go directly on the CRATE_USER. It's true, that this user is the only superuser yet, but i think it's better to get the user from the user list and then check the `isSuperuser()` method